dss/config.py

```diff
@@ -116,32 +116,42 @@ def set_config(config: BucketConfig):
 
     @staticmethod
     @functools.lru_cache()
-    def get_native_handle(replica: "Replica",
-                          connect_timeout: float=None,
-                          read_timeout: float=None) -> typing.Any:
-
-        if connect_timeout is None:
-            connect_timeout = Config.BLOBSTORE_CONNECT_TIMEOUT
-        if read_timeout is None:
-            read_timeout = Config.BLOBSTORE_READ_TIMEOUT
-
+    def get_native_handle(replica: "Replica") -> typing.Any:
         if replica == Replica.aws:
+            return Config._get_native_aws_handle()
+        elif replica == Replica.gcp:
+            return Config._get_native_gcp_handle()
+        raise NotImplementedError(f"Replica `{replica.name}` is not implemented!")
+
+    @staticmethod
+    def _get_native_aws_handle() -> typing.Any:
+        if Config.BLOBSTORE_CONNECT_TIMEOUT is None or Config.BLOBSTORE_READ_TIMEOUT is None:
```

Comment:
what if `BLOBSTORE_BOTO_RETRIES` is set?

Reply:
Also, this is an _or_, which is kind of weird.

Reply:
My idea here was that you should either provide values for all timeouts, or none of them.

Switched to allowing partial configs.